### PR TITLE
Add notifications

### PR DESCRIPTION
**Pull request by Ndpnt**
Notify on scraping errors
Notify on sanitized document update

_There is no tests for now but I still open this PR to allow to review during my days off. 
As testing interaction with SendInBlue seems to be a heavy work, it would be useful for me to discuss what deserve to be tested._

### CODE REVIEW

File: src/notifier/index.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        documentName: documentTypes[documentId].name,
+        documentId,
+      }));
+    });
+  });
+
+  return Promise.all(listsPromises);
+}
+
+async function generateServiceProviderList({ serviceProviderName, serviceProviderId }) {
```
Comment (MattiSG) [inline src/notifier/index.js:58]: This was really ambiguous for me at first read.

```suggestion
async function generateServiceProviderMailingList({ serviceProviderName, serviceProviderId }) {
```

File: src/notifier/index.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  };
+}
+
+export async function onDocumentScrappingError(serviceProviderId, documentTypeId, error) {
+  const sendParams = {
+    templateId: ERROR_TEMPLATE_ID,
+    params: {
+      "SERVICE_PROVIDER_NAME": serviceProviders[serviceProviderId].serviceProviderName,
+      "DOCUMENT_TYPE": documentTypes[documentTypeId].name,
+      "ERROR_TEXT": `An error has occured when trying to scrape the document:
```
Comment (MattiSG) [inline src/notifier/index.js:83]: > Unnecessary quotes.

File: src/notifier/index.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  const updateListId = await createListIfNotExists(updateListName);
+  const errorListId = await createListIfNotExists(errorListName);
+
+  serviceProvidersMailingLists[serviceProviderId][documentId] = {
+    updateListId,
+    errorListId
+  };
+}
+
+export async function onDocumentScrappingError(serviceProviderId, documentTypeId, error) {
```
Comment (MattiSG) [inline src/notifier/index.js:77]:
```suggestion
export async function onDocumentScrapingError(serviceProviderId, documentTypeId, error) {
```

File: src/notifier/index.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  });
+
+  if (!lists) {
+    return [];
+  }
+
+  return _aggregate('getFolderLists', 'lists', folderId, limit, offset + limit, [...lists], count);
+}
+
+async function _aggregate(functionName, resultKey, resourceId, limit, offset, aggregator, count) {
```
Comment (MattiSG) [inline src/notifier/index.js:165]: I don't understand the intent of this whole thing. Rename or document?

**Comment by MattiSG** [comment]
(edit: I did review the entier `notifier.js` file)

File: .env.example [hunk 5]
```diff
@@ -1,2 +1,8 @@
 AUTHOR_NAME='AmbNum Bot'
 AUTHOR_EMAIL='ambnum.bot@disinfo.quaidorsay.fr'
+SENDINBLUE_API_KEY='xkeysib-3f51c…'
+LIST_FOLDER_ID = 44
+LIST_TEST_FOLDER_ID = 184
+UPDATE_TEMPLATE_ID = 21
+ERROR_TEMPLATE_ID = 22
+BASE_URL = "github.com/ambanum/CGUs-data/commit/"
```
[the remark this answers is not in the document]
Comment (Ndpnt) [reply .env.example:8]: It's done on the template side as sendInBlue remove the double dot `:` and if we escape it, it will let the anti slash`\:`.

**Comment by Ndpnt** [comment]
As discussed, this implementation is to complex for actual needs, see #30 instead.

**Pull request closed without merging**